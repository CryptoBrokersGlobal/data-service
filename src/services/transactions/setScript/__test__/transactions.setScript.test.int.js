const createService = require('..');
const { createPgDriver } = require('../../../../db');

const TX_ID = '8Nwjd2tcQWff3S9WAhBa7vLRNpNnigWqrTbahvyfMVrU';
const TX_ID_2 = 'Dt7L51pQbmMcc971byGeBHDgoi7AMfHgCtmoWYMRyWhL';

const loadConfig = require('../../../../loadConfig');
const options = loadConfig();

const drivers = {
  pg: createPgDriver(options),
};

const service = createService({
  drivers,
  emitEvent: () => () => null,
});

describe('SetScript transaction service', () => {
  describe('get', () => {
    it('fetches real tx', async done => {
      service
        .get(TX_ID)
        .run()
        .promise()
        .then(x => {
          expect(x).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
    it('returns null for unreal tx', async () => {
      const tx = await service
        .get('UNREAL')
        .run()
        .promise();

      expect(tx).toBe(null);
    });
  });

  describe('mget', () => {
    it('fetches real txs with nulls for unreal', async done => {
      service
        .mget([TX_ID, 'UNREAL', TX_ID_2])
        .run()
        .promise()
        .then(xs => {
          expect(xs).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
  });
});
