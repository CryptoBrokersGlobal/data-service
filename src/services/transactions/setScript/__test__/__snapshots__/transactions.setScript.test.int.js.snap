// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetScript transaction service get fetches real tx 1`] = `
Object {
  "__type": "transaction",
  "data": Object {
    "fee": "0.02082496",
    "height": 1190001,
    "id": "8Nwjd2tcQWff3S9WAhBa7vLRNpNnigWqrTbahvyfMVrU",
    "proofs": Array [
      "V45jPG1nuEnwaYb9jTKQCJpRskJQvtkBcnZ45WjZUbVdNTi1KijVikJkDfMNcEdSBF8oGDYZiWpVTdLSn76mV57",
    ],
    "script": "base64:AQQAAAAEaW5hbAIAAAAESW5hbAQAAAAFZWxlbmECAAAAB0xlbnVza2EEAAAABGxvdmUCAAAAC0luYWxMZW51c2thCQAAAAAAAAIJAAEsAAAAAgUAAAAEaW5hbAUAAAAFZWxlbmEFAAAABGxvdmV4ZFt5",
    "sender": "3PBSduYkK7GQxVFWkKWMq8GQkVdAGX71hTx",
    "senderPublicKey": "3LZmDK7vuSBsDmFLxJ4qihZynUz8JF9e88dNu5fsus5p",
    "timestamp": 2018-09-26T11:51:52.182Z,
    "type": 13,
    "version": 1,
  },
}
`;

exports[`SetScript transaction service mget fetches real txs with nulls for unreal 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "transaction",
      "data": Object {
        "fee": "0.02082496",
        "height": 1190001,
        "id": "8Nwjd2tcQWff3S9WAhBa7vLRNpNnigWqrTbahvyfMVrU",
        "proofs": Array [
          "V45jPG1nuEnwaYb9jTKQCJpRskJQvtkBcnZ45WjZUbVdNTi1KijVikJkDfMNcEdSBF8oGDYZiWpVTdLSn76mV57",
        ],
        "script": "base64:AQQAAAAEaW5hbAIAAAAESW5hbAQAAAAFZWxlbmECAAAAB0xlbnVza2EEAAAABGxvdmUCAAAAC0luYWxMZW51c2thCQAAAAAAAAIJAAEsAAAAAgUAAAAEaW5hbAUAAAAFZWxlbmEFAAAABGxvdmV4ZFt5",
        "sender": "3PBSduYkK7GQxVFWkKWMq8GQkVdAGX71hTx",
        "senderPublicKey": "3LZmDK7vuSBsDmFLxJ4qihZynUz8JF9e88dNu5fsus5p",
        "timestamp": 2018-09-26T11:51:52.182Z,
        "type": 13,
        "version": 1,
      },
    },
    null,
    Object {
      "__type": "transaction",
      "data": Object {
        "fee": "0.01",
        "height": 1190002,
        "id": "Dt7L51pQbmMcc971byGeBHDgoi7AMfHgCtmoWYMRyWhL",
        "proofs": Array [
          "3wPYieN6CriPVzRDCGCqDnPLR5xDDC1RaQVPT85RiftARZobeAEQ1bMp4rwMSHhSPgKyAGDAfFDGkjvBvKVuqqJg",
        ],
        "script": "base64:AQkAAfQAAAADCAUAAAACdHgAAAAJYm9keUJ5dGVzCQABkQAAAAIIBQAAAAJ0eAAAAAZwcm9vZnMAAAAAAAAAAAABAAAAIJCWiNsyagiZio80TlMvhx+LNSWy+s3wvE/jALCRCi1e2ccqiA==",
        "sender": "3P6WfA4qYtkgwVAsWiiB6yaea2X8zyXncJh",
        "senderPublicKey": "EYNuSmW4Adtcc6AMCZyxkiHMPmF2BZ2XxvjpBip3UFZL",
        "timestamp": 2018-09-26T12:55:09.003Z,
        "type": 13,
        "version": 1,
      },
    },
  ],
}
`;
